grakn-client/src/main/java/ai/grakn/remote/concept/RemoteSchemaConcept.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016-2018 Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.remote.concept;
+
+import ai.grakn.concept.Label;
+import ai.grakn.concept.LabelId;
+import ai.grakn.concept.Rule;
+import ai.grakn.concept.SchemaConcept;
+
+import javax.annotation.Nullable;
+import java.util.stream.Stream;
+
+/**
+ * @author Felix Chapman
+ *
+ * @param <Self> The exact type of this class
```

Comment:
`Self` ? That's an interesting convention you introducing

Reply:
Haha, I was so happy when I realised I could do this!

The only problem I can think of is that `Self` (and `MyType`) look like they might be classes. In an IDE this isn't an issue though:
![image](https://user-images.githubusercontent.com/4632588/36535312-1bd2064e-17c1-11e8-8598-b609d9607cce.png)